core/dbt/clients/jinja.py

```diff
@@ -301,6 +319,15 @@ def exception_handler(self) -> Iterator[None]:
             e.stack.append(self.macro)
             raise e
 
+    def _is_statement_under_materailization(self) -> bool:
+        return bool(
+            self.macro.unique_id == "macro.dbt.statement"
```

Comment:
@jtcohen6 with this user can't actually access submit_python_job in custom `statement` macro. Any concern with that?

Reply:
@ChenyuLInx I'm aligned. I'm not familiar with any reason or legitimate use case for an end user to override the `statement` macro — this should really be treated as internal to `dbt-core`.

If we want to loosen this restriction in the future, we can. (E.g. by changing this conditional to check for any macro named `statement`, rather than the `unique_id` containing project name `dbt`.)